tests/unit/onyxClearWebStorageTest.js

```diff
@@ -0,0 +1,62 @@
+import Storage from '../../lib/storage';
+
+const ONYX_KEYS = {
+    DEFAULT_KEY: 'defaultKey',
+};
+jest.mock('../../lib/storage');
+
+describe('Set data while storage is clearing', () => {
+    let connectionID;
+    let Onyx;
+
+    /** @type OnyxCache */
+    let cache;
+
+    beforeAll(() => {
+        Onyx = require('../../index').default;
+        jest.useRealTimers();
+
+        Onyx.init({
+            keys: ONYX_KEYS,
+            registerStorageEventListener: () => {},
+            initialKeyStates: {
+                [ONYX_KEYS.DEFAULT_KEY]: 'default',
+            },
+        });
+        const webStorageClear = Storage.clear;
+        Storage.clear = jest.fn(() => new Promise(resolve => setTimeout(resolve, 20))
+            .then(webStorageClear));
+    });
+
+    // Always use a "fresh" cache instance
+    beforeEach(() => {
+        cache = require('../../lib/OnyxCache').default;
+    });
+
+    afterEach(() => {
+        Onyx.disconnect(connectionID);
+        return Onyx.clear();
+    });
+
+    it('should store merged values when calling merge on a default key after clear', () => {
```

Comment:
Why only test the "`Onyx.clear()` then `Onyx.merge()`" condition with the web storage but testing 2 additional case in the native storage test?

**NAB question:** Is there some way to have them run the same tests just twice and use a different provider each time?